tests/testthat/test-concRespCore.R

```diff
@@ -15,3 +15,93 @@ test_that("concRespCore works", {
   expect_equal(out$ga, 9.59, tolerance = 1e-2)
 
 })
+
+test_that("HTPP global data internal check", {
+
+  skip_on_cran()
+
+  # load necessary data
+  load("~/CompTox-ToxCast-tcplFit2/R/sysdata.rda")
+
+  ## Confirmed with Derik Haggard, for chemical EPAPLT0617C02,
+  ## concentrations above 2.98 uM are removed from curve-fitting
+  ## because higher concentrations cause cytotoxicity above the standard threshold of 50%.
+  ## (i.e. concentration that is below the fitted EC50 value for a chemical is removed.)
+
+  ## dropping higher concentrations for EPAPLT0617C02
+  htpp_global_input = subset(htpp_global_input, htpp_global_input$chem_id !=  "EPAPLT0617C02" |
+                      htpp_global_input$conc <= 2.98 )
+
+  ## One can use the following code to verify that for curve fitting, EPAPLT0617C02 uses
+  ## 5 unique concentrations, with the highest concentration being 2.98.
+  ## htpp_global_subset[htpp_global_subset$chem_id == "EPAPLT0617C02",c("chem_id", "min_conc", "max_conc","n_conc", "conc")]
+  ## unique(htpp_global_input[htpp_global_input$chem_id == "EPAPLT0617C02","conc"])
+
+
+  my_global <- NULL
+  for (this.chem in unique(htpp_global_subset$chem_id)){
+    ## for each chemical, prepare data for concRespCore
+    temp <- htpp_global_input |> filter(chem_id == this.chem)
+    ## prepare "row" list object per Description
+    row <- list(
+      pg_id = unique(temp$pg_id),
+      stype = unique(temp$stype),
+      chem_id = this.chem,
+      min_conc = min(temp$conc),
+      max_conc = max(temp$conc),
+      n_conc = length(unique(temp$conc)),
+      ctr_mean = CONTROL_GMAH$BMED,
+      ctr_sd = CONTROL_GMAH$CUTOFF,
+      conc = temp$conc,
+      resp = temp$d,
+      bmed = CONTROL_GMAH$BMED,
+      cutoff = CONTROL_GMAH$CUTOFF,
+      onesd = CONTROL_GMAH$ONESD,
+      approach = "global",
+      endpoint = "global"
+    )
+    ## run concRespCore on the ‘row’ list object
+    newLine <- try(concRespCore(row, fitmodels = c("cnst", "hill",  "poly1", "poly2", "pow", "exp2", "exp3","exp4", "exp5"), conthits = T, aicc = F, force.fit = FALSE, bidirectional = FALSE, AUC = FALSE))
+    if(is.null(newLine) | class(newLine)=="try-error"){
+      newLine <- try(concRespCore(row, conthits = F, aicc = FALSE, force.fit = FALSE,  bidirectional = TRUE, fitmodels=c("cnst"), AUC = FALSE))
+    }else if(class(newLine) == "try-error"){
```

Comment:
Not quite sure why you need this logical check since the if statement above should catch the cases where newLine is a try-error.  Please explicate why this is needed if I am missing something.

Reply:
To answer this comment and the comment about doing two fits - the fitting codes come from the Description.doc file sent along with the data files. But in the file it doesn't explicate why. My understanding of the second fit (with only constant, di-direction = T and conthits = F) is an attempt to at least get a baseline model after the first fit failed. We can discuss this with Derik and Josh.